ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionengine/ExecutionEngineChannelMock.java

```diff
@@ -231,4 +253,38 @@ private HeadAndAttributes(Bytes32 head, PayloadAttributes attributes) {
       this.attributes = attributes;
     }
   }
+
+  private void prepareTransitionBlocks() {
+    final SpecVersion specVersion = spec.forMilestone(SpecMilestone.BELLATRIX);
+    checkNotNull(specVersion, "Bellatrix must be scheduled to for transition emulation");
+    final SpecConfigBellatrix specConfigBellatrix =
+        specVersion.getConfig().toVersionBellatrix().orElseThrow();
+
+    final UInt64 bellatrixActivationSlot =
+        spec.computeStartSlotAtEpoch(specConfigBellatrix.getBellatrixForkEpoch());
+
+    // assumes genesis happen immediately at min genesis time reached
+    final UInt64 bellatrixActivationTime =
+        spec.getSlotStartTime(
+            bellatrixActivationSlot, spec.getGenesisSpecConfig().getMinGenesisTime());
```

Comment:
Maybe its easier to implement the transition via terminal block hash

Reply:
Yeah you'll need the state realistically - we may want to use the mock with things other than interop mode anyway.  Hacky work around would be to just have the transition time hard coded.  Or just have a delay after the current time before the transition block is found (the timestamp just needs to be before the transition slot and teku should handle that already because the PoW chain timestamps aren't lined up with the beacon chain at all anyway).

Reply:
ahh.. yeah.. right... I actually don't even need to know when bellatrix activates.
I can use the first `getPowChainHead` call as the bellatrix activation signal. And i can apply a delay from there.